src/server/object_services/md_store.js

```diff
@@ -496,6 +511,16 @@ class MDStore {
         };
     }
 
+    async find_unreclaimed_objects(limit) {
+        const results = await this._objects.col().find({
```

Comment:
On second thought, we should create a separate index for this.
This should be a simple index with `fields: { deleted: 1 }` and `partialFilterExpression: { deleted: { $exists: true }, reclaimed: null }` and we probably want to make sure to give a hint to use it here.

Reply:
The reason is that this "deleted but not yet reclaimed" list should usually be very small.
So there is no real overhead of this index size, and only deletions will insert to the index.